source/codegen/metadata/nisync/functions.py

```diff
@@ -1,761 +1,1499 @@
 # -*- coding: utf-8 -*-
 functions = {
     "init": {
-        'init_method': True,
+        "init_method": True,
         "parameters": [
-            {"direction": "in", "name": "resourceName", "type": "ViRsrc"},
-            {"direction": "in", "name": "idQuery", "type": "ViBoolean"},
-            {"direction": "in", "name": "resetDevice", "type": "ViBoolean"},
-            {"direction": "out", "name": "vi", "type": "ViSession"},
+            {
+                "direction": "in",
+                "name": "resourceName",
+                "type": "ViRsrc",
+            },
+            {
+                "direction": "in",
+                "name": "idQuery",
+                "type": "ViBoolean",
+            },
+            {
+                "direction": "in",
+                "name": "resetDevice",
+                "type": "ViBoolean",
+            },
+            {
+                "direction": "out",
+                "name": "vi",
+                "type": "ViSession",
+            },
         ],
         "returns": "ViStatus",
     },
     "close": {
         "parameters": [
-            {"direction": "in", "name": "vi", "type": "ViSession"},
+            {
+                "direction": "in",
+                "name": "vi",
+                "type": "ViSession",
+            },
         ],
         "returns": "ViStatus",
     },
-    'RevisionQuery':{
-        'cname' : 'niSync_revision_query',
-        'parameters':[
+    "error_message": {
+        "parameters": [
             {
-                'name':'vi',
-                'direction':'in',
-                'type':'ViSession'
+                "direction": "in",
+                "name": "vi",
+                "type": "ViSession",
             },
             {
-                'name':'driverRevision',
-                'direction':'out',
-                'type':'ViChar[]',
-                'size':{
-                    'mechanism':'fixed',
-                    'value':256
-                }
+                "direction": "in",
+                "name": "errorCode",
+                "type": "ViStatus",
             },
             {
-                'name':'firmwareRevision',
-                'direction':'out',
-                'type':'ViChar[]',
-                'size':{
-                    'mechanism':'fixed',
-                    'value':256
-                }
-            }
+                "direction": "out",
+                "name": "errorMessage",
+                "type": "ViChar[]",
+                "size": {
+                    "mechanism": "fixed",
+                    "value": 256,
+                },
+            },
         ],
-        'returns':'ViStatus'
+        "returns": "ViStatus",
     },
-    "SendSoftwareTrigger": {
+    "reset": {
```

Comment:
[Optional] This is up to you and could be done in a future PR.

If you want the gRPC method names in the proto and the C API method name to differ (for example: Reset and reset) you can do something similar to what was done for NIScope here: https://github.com/ni/grpc-device/blob/main/source/codegen/metadata/niscope/functions.py#L2309

Reply:
Yes, I would like that. I added it to the back log.